bpf/mb_connect.c

```diff
@@ -67,6 +67,30 @@ static inline int tcp4_connect(struct bpf_sock_addr *ctx)
         // we only deal pod's traffic managed by istio.
         return 1;
     }
+    __u32 curr_pod_ip = 0;
+    {
+        // get ip addresses of current pod/ns.
+        __u32 curr_ip_mark = 0;
+        struct bpf_sock_tuple tuple = {};
+        tuple.ipv4.dport = bpf_htons(SOCK_IP_MARK_PORT);
+        tuple.ipv4.daddr = 0;
+        struct bpf_sock *s = bpf_sk_lookup_tcp(ctx, &tuple, sizeof(tuple.ipv4),
+                                               BPF_F_CURRENT_NETNS, 0);
+        if (s) {
+            curr_ip_mark = s->mark;
+            bpf_sk_release(s);
+            __u32 *ip = bpf_map_lookup_elem(&mark_pod_ips_map, &curr_ip_mark);
+            if (ip) {
+                curr_pod_ip = *ip; // network order
+            } else {
+                debugf("get ip for mark %x error", curr_ip_mark);
+            }
+        }
+    }
+    if (curr_pod_ip == 0) {
+        debugf("get current pod ip error");
+        return 1;
```

Comment:
So for the pods running before merbridge, the dst port will never get rewrited?

We have 2 solutions to solve this:
1. use the legacy logic, then we should not return at this point
2. listen on `SOCK_IP_MARK_PORT` in all injected pods when merbridge starts, then I think we can remove the legacy logic

Reply:
error logic. hhh. copy code error.